syntax/forth.vim

```diff
@@ -181,36 +180,28 @@ syn keyword forthBlocks BLOCK-INCLUDED BLK
 " numbers
 syn keyword forthMath DECIMAL HEX BASE
 syn match forthInteger '\<-\=[0-9]\+.\=\>'
-syn match forthInteger '\<&-\=[0-9]\+.\=\>'
 syn match forthInteger '\<#-\=[0-9]\+.\=\>'
-" recognize hex and binary numbers, the '$' and '%' notation is for gforth
-syn match forthInteger '\<\$\x*\x\+\>' " *1* --- don't mess
-syn match forthInteger '\<\x*\d\x*\>'  " *2* --- this order!
-syn match forthInteger '\<%[0-1]*[0-1]\+\>'
-syn match forthFloat '\<-\=\d*[.]\=\d\+[DdEe]\d\+\>'
-syn match forthFloat '\<-\=\d*[.]\=\d\+[DdEe][-+]\d\+\>'
+syn match forthInteger '\<\$-\=\x*\x\+\>'
+syn match forthInteger '\<%-\=[0-1]*[0-1]\+\>'
+syn match forthFloat '\<[+-]\=\d*[.]\=\d*[DdEe][+-]\=\d*\>'
 
 " XXX If you find this overkill you can remove it. this has to come after the
 " highlighting for numbers otherwise it has no effect.
 syn region forthComment start='0 \[if\]' end='\[endif\]' end='\[then\]' contains=forthTodo
 
 " Strings
-syn region forthString start=+\.*"+ end=+"+ end=+$+ contains=@Spell
-" XXX
-syn region forthString start=+s"+ end=+"+ end=+$+ contains=@Spell
-syn region forthString start=+c"+ end=+"+ end=+$+ contains=@Spell
-
+syn region forthString matchgroup=forthString start=+s"\s+ end=+"+ end=+$+ contains=@Spell
+syn region forthString matchgroup=forthString start=+c"\s+ end=+"+ end=+$+ contains=@Spell
+syn region forthString matchgroup=forthString start=+\."\s+ end=+"+ end=+$+ contains=@Spell
 syn region forthString matchgroup=forthString start=+s\\"+ end=+"+ end=+$+ contains=@Spell,forthEscape
 
 syn match forthEscape +\\[abeflmnqrtvz"\\]+ contained
 syn match forthEscape "\\x\x\x" contained
 
 " Comments
 syn match forthComment '\\\%(\s.*\)\=$' contains=@Spell,forthTodo,forthSpaceError
-syn region forthComment start='\\S\s' end='.*' contains=@Spell,forthTodo,forthSpaceError
```

Comment:
I have no idea where `\S` is from. It is not a standard word.

Reply:
This is often used to comment to EOF.  E.g. https://github.com/philburk/pforth/blob/8da2f097a6a4b856e659e08556ff938ac612d8ed/fth/misc1.fth#L179

However, I don't think there's much point in supporting these non-standard features in an ad hoc manner.  I'd remove it.

Ideally support for other dialects would be built on top of a standard Forth filetype.

Reply:
Oh! It seems like it should be removed, especially if different Forths use it for completely different things.